File: frontend/types/order.ts
export type OrderStatus = "DRAFT" | "SUBMITTED" | "ACCEPTED" | "PREPARING" | "READY" | "COMPLETED" | "CANCELLED";
export type Money = number | string;

export interface OrderItem {
    order_item_id: number;
    order_id: number;
    menu_item_id: number | null;
    menu_item_name: string;
    quantity: number;
    unit_price: Money;
    line_total: Money;
    notes: string | null;
}

export interface Order {
    order_id: number;
    restaurant_id: number;
    table_number: string | null;
    customer_name: string | null;
    status: OrderStatus;
    payment_status: "UNPAID" | "PAID" | "VOID";
    notes: string | null;
    subtotal: Money;
    total: Money;
    created_at: string;
    updated_at: string;
    items: OrderItem[];
}

export interface OrderUpdate {
    table_number?: string | null;
    customer_name?: string | null;
    status?: OrderStatus;
    notes?: string | null;
    payment_status?: "UNPAID" | "PAID" | "VOID";
}

export interface OrderCreateItem {
    menu_item_id: number;
    quantity: number;
    notes?: string;
}

export interface OrderCreate {
    table_number?: string;
    customer_name?: string;
    notes?: string;
    items: OrderCreateItem[];
}

export interface PaginatedOrders {
    items: Order[];
    total: number;
    limit: number;
    offset: number;
}

File: backend/app/schemas/order.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class OrderItemCreate(BaseModel):
    menu_item_id: int
    quantity: int = Field(gt=0)
    notes: str | None = None


class OrderItemResponse(BaseModel):
    order_item_id: int
    order_id: int
    menu_item_id: int
    menu_item_name: str
    quantity: int
    unit_price: Decimal
    line_total: Decimal
    notes: str | None = None

    model_config = ConfigDict(from_attributes=True)


class OrderCreate(BaseModel):
    table_number: str | None = None
    customer_name: str | None = None
    notes: str | None = None
    items: list[OrderItemCreate] = Field(min_length=1)


class OrderUpdate(BaseModel):
    table_number: str | None = None
    customer_name: str | None = None
    status: str | None = None
    notes: str | None = None
    items: list[OrderItemCreate] | None = Field(default=None, min_length=1)


class OrderResponse(BaseModel):
    order_id: int
    restaurant_id: int
    table_number: str | None = None
    customer_name: str | None = None
    status: str
    notes: str | None = None
    subtotal: Decimal
    total: Decimal
    created_at: datetime
    updated_at: datetime
    items: list[OrderItemResponse]

    model_config = ConfigDict(from_attributes=True)
